OptimizelySDK/Bucketing/DecisionService.cs

```diff
@@ -289,12 +289,12 @@ public virtual Variation GetVariationForFeatureRollout(FeatureFlag featureFlag,
             // For all rules before the everyone else rule
             for (int i=0; i < rolloutLength - 1; i++)
             {
-                var rolloutExperiment = rollout.Experiments[i];
+                var rolloutRule = rollout.Experiments[i];
 
-                if (ExperimentUtils.IsUserInExperiment(ProjectConfig, rolloutExperiment, filteredAttributes))
+                if (ExperimentUtils.IsUserInExperiment(ProjectConfig, rolloutRule, filteredAttributes))
                 {
-                    Logger.Log(LogLevel.DEBUG, $"Attempting to bucket user \"{userId}\" into rollout rule \"{rolloutExperiment.Key}\".");
-                    variation = Bucketer.Bucket(ProjectConfig, rolloutExperiment, userId, userId);
+                    Logger.Log(LogLevel.DEBUG, $"Attempting to bucket user \"{userId}\" into rollout rule \"{rolloutRule.Key}\".");
+                    variation = Bucketer.Bucket(ProjectConfig, rolloutRule, userId, userId);
```

Comment:
we should evaluate bucketingId from the userAttributes and then use it here.

Reply:
Awesome catch of this pull request. Fixed now.